### PR TITLE
Add active_record backend

### PR DESCRIPTION
**Pull request by gottfrois**
Needs specs if we go for it

### CODE REVIEW

File: rails_event_store_active_record.gemspec [hunk 1]
```diff
@@ -1,32 +1,24 @@
 # coding: utf-8
-lib = File.expand_path('../lib', __FILE__)
-$LOAD_PATH.unshift(lib) unless $LOAD_PATH.include?(lib)
-require 'rails_event_store_active_record/version'
 
 Gem::Specification.new do |spec|
   spec.name          = "rails_event_store_active_record"
-  spec.version       = RailsEventStoreActiveRecord::VERSION
+  spec.version       = "1.0.0"
```
Comment (gottfrois) [inline rails_event_store_active_record.gemspec]: this version should probably mirror rails_event_store version (`0.2.0` currently)

**Comment by mpraglowski** [comment]
PR https://github.com/arkency/rails_event_store/pull/21 should be merged here - it fixes issue reported when reading all streams.

**Comment by mpraglowski** [comment]
@gottfrois this repository now has latest version of code from RES - could you rebase & update this PR ?

**Comment by gottfrois** [comment]
@mpraglowski cherry-picked latest event repository code from rails_event_store. Didn't find specs for the adapter though, did I miss it?

**Comment by mpraglowski** [comment]
> cherry-picked latest event repository code from rails_event_store.
Why ? master here has already the latest code form RES

> Didn't find specs for the adapter though, did I miss it?
Spec for event repository is in ruby_event_store (kind of contract for all future event repository implementations) - see it here https://github.com/arkency/ruby_event_store/blob/master/lib/ruby_event_store/spec/event_repository_lint.rb

All the spec for implementation of event repository should be like https://github.com/arkency/rails_event_store_active_record/blob/master/spec/event_repository_spec.rb#L6

**Comment by mpraglowski** [comment]
Current implementation passes this tests.

**Comment by gottfrois** [comment]
> Why ? master here has already the latest code form RES

I mean, I did it and force pushed :)

> All the spec for implementation of event repository should be like

Ok i'll have a look at the specs

**Comment by mpraglowski** [comment]
Closing this PR - too much changes - no need for it here - IMHO all changes required are in RES

**Comment by gottfrois** [comment]
Have you thought on how you want to handle changing store adapter? This PR assumed the following API:

```ruby
RailsEventStore::Repository.backend = :active_record
```

**Comment by mpraglowski** [comment]
Use new configuration for RES

```ruby
RailsEventStore.configure do |config|
  config.event_repository = YourOwnEventRepository.new
end

# and then
client = RailsEventStore::Client.new
```

or just pass it as argument:

```ruby
client = RailsEventStore::Client.new(repository: YourOwnEventRepository.new)
```

Check this commit in RES https://github.com/arkency/rails_event_store/commit/315a385189d347f16a2c5c132263b0fe384e2298

**Pull request closed without merging**